server/core/src/main/java/org/infinispan/server/core/transport/ConnectionMetadata.java

```diff
@@ -0,0 +1,89 @@
+package org.infinispan.server.core.transport;
+
+import java.net.SocketAddress;
+import java.time.Instant;
+
+import io.netty.channel.Channel;
+import io.netty.util.AttributeKey;
+
+/**
+ * @since 15.0
+ **/
+public class ConnectionMetadata {
+   static final AttributeKey<Long> ID = AttributeKey.newInstance("ID");
+   static final AttributeKey<Instant> CREATED = AttributeKey.newInstance("CREATED");
+   static final AttributeKey<String> CLIENT_PRINCIPAL = AttributeKey.newInstance("CLIENT_PRINCIPAL");
+   static final AttributeKey<String> CLIENT_NAME = AttributeKey.newInstance("CLIENT_NAME");
+   static final AttributeKey<String> CLIENT_LIBNAME = AttributeKey.newInstance("CLIENT_LIBRARY_NAME");
+   static final AttributeKey<String> CLIENT_LIBVER = AttributeKey.newInstance("CLIENT_LIBRARY_VERSION");
+   static final AttributeKey<String> CLIENT_PROTOCOL = AttributeKey.newInstance("CLIENT_PROTOCOL");
+   private final Channel channel;
+
+   public ConnectionMetadata(Channel channel) {
```

Comment:
I would just say to have this as class methods instead of instance methods. Seems we are just creating an object for no reason otherwise.

Reply:
That's what I've done now